pkg/virt-launcher/notify-client/client.go

```diff
@@ -196,12 +196,13 @@ func newWatchEventError(err error) watch.Event {
 }
 
 func eventCallback(c cli.Connection, domain *api.Domain, libvirtEvent libvirtEvent, client *Notifier, events chan watch.Event,
-	interfaceStatus []api.InterfaceStatus, osInfo *api.GuestOSInfo) {
+	interfaceStatus []api.InterfaceStatus, osInfo *api.GuestOSInfo, domainName string) {
```

Comment:
I think that with so many arguments, this merits a struct already. Everyone is adding "just" one and it becomes too big to manage/maintain/read.

Maybe you can add a small refactor commit to fix it.

Reply:
@oshoval Since *api.Domain already passed to the function I think you can invoke domain name from api.Domain.Spec

Reply:
@enp0s3 
Thanks but i am afraid it can be nil
see please
https://github.com/kubevirt/kubevirt/blob/2d22cb6a9706e0ef0e1a9cc7b0b7fd1d57480730/pkg/virt-launcher/notify-client/client.go#L333

Reply:
@oshoval How come? here the domain cache is being created: https://github.com/kubevirt/kubevirt/blob/2d22cb6a9706e0ef0e1a9cc7b0b7fd1d57480730/pkg/virt-launcher/notify-client/client.go#L320
If it should be thread-safe lets protect it with lock, otherwise how does it become nil?

Reply:
its not about thread safe, 
its about 
https://github.com/kubevirt/kubevirt/blob/2d22cb6a9706e0ef0e1a9cc7b0b7fd1d57480730/pkg/virt-launcher/notify-client/client.go#L333

there are three select cases there in one infinite loop, the first channel that will happen will execute (in case two at the same time one is taken randomly, this is the select logic of go), and see please that this case even check its not nil